sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/TestEnvironment.cs

```diff
@@ -154,38 +194,114 @@ private static string ReadAndVerifyEnvironmentVariable(string variableName)
         }
 
         /// <summary>
-        ///   Parses a well-formed connection string to extract the fully qualified namespace.
+        ///   Reads an environment variable.
         /// </summary>
         ///
-        /// <param name="connectionString">The connection string to parse.</param>
+        /// <param name="variableName">The name of the environment variable to read.</param>
         ///
-        /// <returns>The fully qualified namespace contained in the connection string.</returns>
+        /// <returns>The value of the environment variable, if present and populated; null otherwise</returns>
         ///
-        private static string ParseFullyQualifiedNamespace(string connectionString)
-        {
-            var start = connectionString.IndexOf("Endpoint");
+        private static string ReadEnvironmentVariable(string variableName) =>
```

Comment:
Can we slide this down so that it has locality with `ReadAndVerifyEnvironmentVariable`?